File: lesunny/062_멀쩡한사각형.java
class Solution {
    int gcd(int a, int b){
        if(b==0)
            return a;
        else
            return gcd(b,a%b);
    }
    public long solution(int w, int h) {
        long answer = 1;
        int gcd = 0;
        if(w>=h)
            gcd = gcd(w,h);
        else
            gcd = gcd(h,w);        
        
        answer = (long)w*h -w - h +gcd;
        
        return answer;
    }
}

File: JIWON/062 멀쩡한 사각형.cpp
#include <algorithm>

using namespace std;

long long solution(int w,int h) {
    long long answer = 1;
    long long gcd = 0;
    long long sum = (long long)w*h;
    
    if(w > h) swap(w, h);
    
    for(int i = w; i > 0; i--){
        if(w % i == 0 && h % i == 0){
            gcd = i;
            break;
        }
    }
    
    answer = sum - gcd*((w/gcd) + (h/gcd) - 1);
  
    return answer;
}
